js/index.js

```diff
@@ -1,33 +1,49 @@
 // ITERATION 1
 
-function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
 
-  //... your code goes here
+function updateSubtotal(product) {
+  
+  const price = product.querySelector('.price span').innerHTML 
+
+  const quantity = product.querySelector(".quantity input").value
+  const subtotal = Number(price) * Number(quantity);
+  const subtotalElement = product.querySelector(".subtotal span")
+  subtotalElement.innerHTML = subtotal;
+  return subtotal 
 }
 
+
+
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
 
-  // ITERATION 2
-  //... your code goes here
+
+  
+   const cartItemElements = document.getElementsByClassName('product');
+
+  let totalValue = 0;
+
+  for(let cartItemElement of cartItemElements) {
+    totalValue += updateSubtotal(cartItemElement);
+  }
+ 
+
+
 
   // ITERATION 3
-  //... your code goes here
+  document.querySelector("#total-value span").innerHTML = totalValue
 }
 
-// ITERATION 4
-
 function removeProduct(event) {
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
-}
 
+  const productRow = target.parentNode.parentNode;
+
+  const productTable = productRow.parentNode;
+
+  productTable.removeChild(productRow);
+
+  calculateAll();
+}
 // ITERATION 5
 
 function createProduct() {
```

Comment:
Good work, I don't see anything worth mentioning with your code(maybe use less whitespace )

I will add a working solution for iteration 5, let me know if you have any questions
```
function createProduct(event) {
  const createRow = document.querySelector('.create-product');
  let newProdNameInput = createRow.querySelector('input');
  let newProdNameValue = newProdNameInput.value;
  let newProdPriceInput = createRow.querySelector("input[type='number']");
  let newProdPriceValue = Number(newProdPriceInput.valueAsNumber).toFixed(2);

  const newTableRow = document.createElement('tr');
  newTableRow.className = 'product';
  newTableRow.innerHTML = `
    <td class="name">
      <span>${newProdNameValue}</span>
    </td>
    <td class="price">$<span>${newProdPriceValue}</span></td>
    <td class="quantity">
      <input type="number" value="0" min="0" placeholder="Quantity" />
    </td>
    <td class="subtotal">$<span>0</span></td>
    <td class="action">
      <button class="btn btn-remove">Remove</>
    </td>
  `;

  // get the parent of this newly created row
  const parent = document.querySelector('#cart tbody');

  // append the newly created row to the parent
  parent.appendChild(newTableRow);

  // make sure remove button inherits the same behavior as other remove buttons
  const removeBtn = newTableRow.querySelector('.btn-remove');
  removeBtn.addEventListener('click', removeProduct);

  // clean the fields
  newProdNameInput.value = '';
  newProdPriceInput.value = 0;
}